### PR TITLE
Documentation for basic usage of plugin

### PR DESCRIPTION
**Pull request by dai-chen**
*Issue #, if available:* https://github.com/opendistro-for-elasticsearch/sql/issues/293, https://github.com/opendistro-for-elasticsearch/sql/issues/243

*Description of changes:* This is the first PR for the documentation improvement work. In this PR, basic usage of our plugin is documented including endpoint, interfaces, plugin monitoring and settings. The generated documentation could be processed and polished further if needed.

*Documentation:* For what the documentation generated looks like, please refer to https://github.com/dai-chen/sql/blob/improve-documentation-basics/docs/user/index.rst

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/user/interfaces/endpoint.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	  },
+	  "_source" : {
+	    "includes" : [
+	      "firstname",
+	      "lastname"
+	    ],
+	    "excludes" : [ ]
+	  }
+	}
+
```
Comment (chloe-zh) [inline docs/user/interfaces/endpoint.rst]: Do we need to add the instruction of `PUT` endpoint?

File: docs/user/index.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (chloe-zh) [inline docs/user/index.rst]: Will this manual entrance be put onto the README file? I think that would be an effective way for user to see refer to the manual.

File: docs/user/admin/monitoring.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+.. contents::
+   :local:
+   :depth: 1
+
+
+Introduction
+============
+
+By a stats endpoint, you are able to collect metrics for the plugin within the interval. Note that only node level statistics collecting is implemented for now. In other words, you only get the metrics for the node you're accessing. Cluster level statistics is yet to implement.
```
Comment (davidcui-amzn) [inline docs/user/admin/monitoring.rst]: Not sure how strict we want to be on correct grammar and semantics- but in the interest of insisting on the highest standards, the last sentence should be worded something like "Cluster level statistics have yet to be implemented."

File: docs/user/interfaces/endpoint.rst (hunk 1 above)
Comment (davidcui-amzn) [reply docs/user/interfaces/endpoint.rst]: +1, I don't see a TODO for adding `PUT` or `DELETE` requests which I think should be included

File: docs/user/index.rst (hunk 2 above)
Comment (dai-chen) [reply docs/user/index.rst]: Sure, I forgot to do that. :) Will update README. Thanks!

File: docs/user/interfaces/endpoint.rst (hunk 1 above)
Comment (dai-chen) [reply docs/user/interfaces/endpoint.rst]: We only support `PUT` and `POST`: https://github.com/opendistro-for-elasticsearch/sql/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java#L83. User needs to `PUT` against Elasticsearch endpoint instead.

File: docs/user/admin/monitoring.rst (hunk 3 above)
Comment (dai-chen) [reply docs/user/admin/monitoring.rst]: Sure, will do. Thanks!

**Comment by dai-chen** [comment]
@aetter it'd be awesome if you can help take a quick look at the 4 documents in this PR, especially the description in each Section and Example. Because we're planning to push these generated docs to `for-elasticsearch-docs` repo later, it would save the trouble to come back to change to code again. Here is the [index page](https://github.com/dai-chen/sql/blob/improve-documentation-basics/docs/user/index.rst). Thanks in advance!

Btw, do you see any problem with these ReStructure Text docs (rather than Markdown) to be pushed to `for-elasticsearch-docs` later?

**Comment by aetter** [comment]
Hi @dai-chen, yeah, I love reStructuredText, but its big appeal is for use with [Sphinx](http://www.sphinx-doc.org/) in large, harder-to-manage sets of documentation. If you're just rendering content on GitHub, it loses a lot of its appeal. Many directives don't work, no [cross-references](http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations), no [transclusion](http://docutils.sourceforge.net/docs/ref/rst/directives.html#including-an-external-document-fragment), no [conditional processing](http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#including-content-based-on-tags), etc.

Our current documentation site uses Jekyll, which doesn't support RST out of the box. Even if we got RST working through a plugin, we'd have to add in the metadata that Jekyll uses to categorize/order/index the content at build time. You can see an example [here](https://raw.githubusercontent.com/opendistro/for-elasticsearch-docs/master/docs/sql/index.md).

So I suspect it would be a lot of effort, and in the end, we'd be left with a documentation site that's 95% Markdown and 5% RST. If the goal is to put these into the documentation site, I'd highly recommend switching to Markdown. [Pandoc](https://pandoc.org/) is super useful for moving between document formats. The syntax is something like:

```bash
pandoc myfile.rst -f rst -t markdown_github -o myfile.md
```

If the reason for choosing RST was that Markdown has some feature gaps, we use the [kramdown](https://kramdown.gettalong.org/syntax.html) parser, which closes a few of them (with some questionable syntax choices). You can get an autogenerated table of contents, for example, like so:

```
#### Table of contents
1. TOC
{:toc}
```

I don't see anything else in your content that uses RST-specific features, so I suspect the conversion to Markdown should be fairly clean if you decide to go that route.

**Comment by dai-chen** [comment]
> Hi @dai-chen, yeah, I love reStructuredText, but its big appeal is for use with [Sphinx](http://www.sphinx-doc.org/) in large, harder-to-manage sets of documentation. If you're just rendering content on GitHub, it loses a lot of its appeal. Many directives don't work, no [cross-references](http://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations), no [transclusion](http://docutils.sourceforge.net/docs/ref/rst/directives.html#including-an-external-document-fragment), no [conditional processing](http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#including-content-based-on-tags), etc.
> 
> Our current documentation site uses Jekyll, which doesn't support RST out of the box. Even if we got RST working through a plugin, we'd have to add in the metadata that Jekyll uses to categorize/order/index the content at build time. You can see an example [here](https://raw.githubusercontent.com/opendistro/for-elasticsearch-docs/master/docs/sql/index.md).
> 
> So I suspect it would be a lot of effort, and in the end, we'd be left with a documentation site that's 95% Markdown and 5% RST. If the goal is to put these into the documentation site, I'd highly recommend switching to Markdown. [Pandoc](https://pandoc.org/) is super useful for moving between document formats. The syntax is something like:
> 
> ```shell
> pandoc myfile.rst -f rst -t markdown_github -o myfile.md
> ```
> 
> If the reason for choosing RST was that Markdown has some feature gaps, we use the [kramdown](https://kramdown.gettalong.org/syntax.html) parser, which closes a few of them (with some questionable syntax choices). You can get an autogenerated table of contents, for example, like so:
> 
> ```
> #### Table of contents
> 1. TOC
> {:toc}
> ```
> 
> I don't see anything else in your content that uses RST-specific features, so I suspect the conversion to Markdown should be fairly clean if you decide to go that route.

Hi @aetter, thanks for the information! The main reason (apart from its nicer looking) I use ReST is to keep the flexibility to integrate with other tool like Sphinx as you mentioned and Python DocTest etc. And yes, I found GitHub itself has very limited support so I avoid some directives for now :). Will convert and push Markdown version of the docs to `for-elasticsearch-docs` later. Thanks!

**Comment by dai-chen** [comment]
@aetter Pandoc works for me. I'm sending PR https://github.com/opendistro/for-elasticsearch-docs/pull/146 for your review. Will merge this PR afterwards.

**Pull request merged**